yarn-project/p2p/src/attestation_pool/memory_attestation_pool.ts

```diff
@@ -0,0 +1,70 @@
+import { type BlockAttestation } from '@aztec/circuit-types';
+import { createDebugLogger } from '@aztec/foundation/log';
+
+import { type AttestationPool } from './attestation_pool.js';
+
+export class InMemoryAttestationPool implements AttestationPool {
+  private attestations: Map</*slot=*/ bigint, Map</*address=*/ string, BlockAttestation>>;
+
+  constructor(private log = createDebugLogger('aztec:attestation_pool')) {
+    this.attestations = new Map();
+  }
+
+  public getAttestationsForSlot(slot: bigint): Promise<BlockAttestation[]> {
+    const slotAttestationMap = this.attestations.get(slot);
+    if (slotAttestationMap) {
+      return Promise.resolve(Array.from(slotAttestationMap.values()));
+    } else {
+      return Promise.resolve([]);
+    }
+  }
+
+  public async addAttestations(attestations: BlockAttestation[]): Promise<void> {
```

Comment:
As mentioned in the test. I think it is slightly weird here that we don't try matching the attestation with what is signed over beyond just taking the slot number as gospel. Haven't seen where it is used yet, but seems like it would be good for us to ensure that we don't end up with a bunch of useless attestations in here.

Reply:
https://github.com/AztecProtocol/aztec-packages/issues/8028